### ISSUE TITLE
mix of ConfirmAjaxButton and IndicatingAjaxButton

### ISSUE THREAD

**Issue by solomax** [sebfz1/wicket-jquery-ui#228]
I need to perform "dangerous, long time" operation

this is why I would like to display confirmation dialog, then if OK was selected start "long operation" and display  "work in progress" indicator

I would love to have this in 7.3.0, but it's not urgent :)

Martin pointed out this can be achieved as follows: http://stackoverflow.com/a/35598376/497381
Unfortunately I was unable to implement it myself :(

**Comment by sebfz1** [comment]
Hi Maxim, 

I've added `ConfirmAjaxButton#isIndicating` so the button becomes indicating (and disabled) when the user clicks the 'ok' dialog button.
I didn't made it for `ConfirmButton` because the implementation will be slightly different and I have to think a little bit more on this...

Furthermore, the behavior differs from the kendo ui implementation, where it is the dialog's 'ok' button that becomes indicating & disabled. I think I prefer this way, because the implementation behind `ConfirmAjaxButton#isIndicating` breaks some principles (although minors). So I do not 100% guarantee I will keep it.

Anyway I've deployed `7.2.2-SNAPSHOT` so you have a chance to test it and give me your feedback :)

Thanks & best regards,
Sebastien.

**Comment by solomax** [comment]
Hello Sebastien,

I just tried new features and here is the feedback :)

1) I was unable to set IndicatingAjaxButton to <button> tag with label because:
<button><wicket:message/></button> is prohibited, IndicatingAjaxButton(id, Model.of("title")) will add "value" attribute which is not being processed

2) ConfirmAjaxButton("cleanup", "label", "title", "message") with <button></button> markup produces additional empty <button> tag

I suppose <input> tag should be used? and <button> is not allowed?

**Comment by sebfz1** [comment]
Hi Maxim,

Right, the model arg of the `Button` stands for the input's value attribute, it does not stand for the button text in case it's a "button" tag. That's a wicket issue (to me) I want to correct since years, but it is not as straightforward as it seems, several questions have to be answered on how/when the model should be applied (for instance, what do we do if the button already contains an underlying wicket:id?).

`AjaxConfirmButton` should be applied on a span or div tag because it embeds a Dialog

By the way, I did manage the ajax-indicator on the dialog's button (as it is in kendo-ui), using martin's link! ;). I need to finalize it before you can test this approach.

Best regards,
Sebastien

**Comment by solomax** [comment]
Thanks!
will wait for new SNAPSHOT to test :)

**Comment by solomax** [comment]
Just perform additional test: IndicatingAjaxButton with <input type="button"/> markup
It is being displayed as expected, but unfortunately "indicating" is not working :(

**Comment by sebfz1** [comment]
Hi Maxim, the icon can only be displayed on a "button" tag...

**Comment by solomax** [comment]
I already get that :(
Maybe tag need to be checked to avoid misuse of this component?

**Comment by sebfz1** [comment]
Hi Maxim,

The problem is that it legal to use XxxButton on both input and button tag.
It's the same problem for jquery-ui & kendo-ui buttons for instance.
In this particular case of the IndicatingAjaxButton, yes maybe I can do a
check and issue a warning...

**Comment by sebfz1** [comment]
Hi Maxim,

I've implemented the busy-indicator on the dialog button itself. It is enabled by default on the ConfirmAjaxButton's dialog, so you have nothing to do except removing your previous changes (I removed the first approach, both cannot coexist)...

This approach is technically better to me, and behave the same as kendo-ui's ConfirmAjaxButton. I hope you will like it! :)

Also, Indicating[Ajax]Button & Secured[Ajax]Button now warns a message if the tag is not `button`...

`7.2.2-SNAPSHOT` has been redeployed.
